python/cudf/cudf/core/scalar.py

```diff
@@ -360,6 +360,9 @@ def _dispatch_scalar_unaop(self, op):
     def astype(self, dtype):
         return Scalar(self.device_value, dtype)
 
+    def _as_host_type(self, dtype):
+        return Scalar(self.value, dtype)
```

Comment:
Instead of introducing this should we update the `astype` function to look at which value is current and use that instead?

Reply:
Moved logic to `astype`. `self.value` will retrieve `host_value` if it does not exist.